src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java

```diff
@@ -2639,4 +2723,29 @@ public static void minimizeWindow(RemoteRobot remoteRobot) {
         keyboard.hotKey(VK_WINDOWS, VK_DOWN);
         keyboard.enter();
     }
+    /**
+     * Handles version-specific menu actions based on the IntelliJ IDEA version.
+     *
+     * @param remoteRobot        Instance of the RemoteRobot to interact with the IntelliJ UI.
+     * @param menuAction2024_2   The submenu option for IntelliJ version 2024.2.
+     * @param menuAction2024_3   The submenu option for IntelliJ version 2024.3.
+     * @throws UnsupportedOperationException if the IntelliJ version is not supported.
+     */
+    public static String handleMenuBasedOnVersion(RemoteRobot remoteRobot, String menuAction2024_2, String menuAction2024_3) {
```

Comment:
Thank you for the feedback @mrglavas . I agree that this might not be an ideal pattern, especially if we plan to support more versions in the future. I'll look into this in detail later to identify a more scalable solution.
Thanks